mongodb/snooty
# DOP-4941: Allow fonts for hidden languages to work without feature flag

### Stories/Links:

DOP-4941

### Current Behavior:

* [ja-jp](https://www.mongodb.com/ja-jp/docs/atlas/getting-started/) - Main issue: `Euclid Circular A` is being used as the font instead of `Noto Sans JP`. This is what's causing the Japanese characters to look way too big in comparison to English characters.
* [zh-cn](https://www.mongodb.com/zh-cn/docs/atlas/getting-started/)
* [ko-kr](https://www.mongodb.com/ko-kr/docs/atlas/getting-started/)
* [pt-br](https://www.mongodb.com/pt-br/docs/atlas/getting-started/)
* [English](https://www.mongodb.com/docs/atlas/getting-started/)

### Staging Links:

* [ja-jp](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/ja-jp/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started-jp/index.html) - Built with no feature flag to show that intended font will still be used, but Japanese will not appear as an option in the consistent-nav components. Japanese text is just for show.
* [zh-cn](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/zh-cn/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [ko-kr](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/ko-kr/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [pt-br](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/pt-br/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [English](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)

**To test:**
1. Go to the Japanese staging link.
2. Ensure that Japanese does **not** show up in consistent-nav language dropdown. (This won't be the case for pre-prd since pre-prd has the feature flag turned on)
3. Ensure that `Noto Sans JP` font is loaded correctly on initial load.
4. Ensure Chinese and Korean sites load in their respective fonts correctly on initial load as well.

### Notes:

* Turns out the reason why font sizes were wonky for Japanese was because `Noto Sans JP` was not being used due to the lack of feature flag. This PR allows the use of a new `getAvailableLanguages` function that can get all languages, either through a feature flag or boolean. I have reached out to our Japanese translator, Saori, who has confirmed that the current font on preprd looks good.
* Fonts for different locales will now be initially loaded in without needing to check for the window's pathname. This prevents the flash of different font that is currently present on localized docs sites ([example](https://www.mongodb.com/zh-cn/docs/atlas/)).
* A new `GATSBY_LOCALE` env variable is now supported. This is used to help modify the `html` tag's `lang` attribute for local development purposes. The `lang` attribute is what the new CSS will use to help ensure correct fonts are used on initial load. Modifying the `lang` locally allows us to mimic how Smartling currently rewrites the `lang` attribute on localized sites. To use, add `GATSBY_LOCALE` to your env files and set it to a locale code.
* No major tests were included since it's a pretty visual change, and Japanese will no longer be a hidden language soon.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README
